### DISCUSSION TITLE
How do contextual queries work

### DISCUSSION THREAD

**Discussion by gavinthomas-valtech** [lukemorales/query-key-factory#44] [Q&A]
Confused slightly about contextual queries. How do they work, whats the use case? to me they look the same as using a nested object. how does it compare to standard react query?

**Comment by lukemorales** [comment]
@gavinthomas-valtech the idea is that sometimes you application might have queries that have a "parent context".

## Example
You have an API call: `api.com/users/:userId/favorites/:favoriteId`, so in order to make that request, you need both `userId` and `favoriteId`, so you would create a nested config in your `usersKeys.contextualQueries`:

```ts
const usersKeys = createQueryKeys('users', {
      detail: (userId: string) => ({
        queryKey: [userId],
        queryFn: () => Promise.resolve({ id: userId }),
        contextQueries: {
          favorite:(favoriteId: string) => { ... },
        },
      }),
    });
```

That way, you can use in `useQuery`:
```ts
useQuery(usersKeys.detail(userId)._ctx.favorite(favoriteId))
```

**Reply by gavinthomas-valtech** [reply]
ah.. So it's not actually making the dependant query. only that you have to have made a detail userQuery first to retreive ```userId```

**Reply by gavinthomas-valtech** [reply]
BTW. is there a codesandbox example? COuld have sworn i saw the pet client one showing contextual queries
